daml-lf/integration-test-lib/src/main/com/daml/lf/CantonRunner.scala

```diff
@@ -0,0 +1,189 @@
+          val ledgerPorts = ports.map(_._2.port)
+          val domainPublicApi = LockedFreePort.find()
+          val domainAdminApi = LockedFreePort.find()
+
+          val cantonPath = rlocation(
+            "external/canton/lib/canton-open-source-2.7.0-SNAPSHOT.jar"
+          )
+          val exe = if (sys.props("os.name").toLowerCase.contains("windows")) ".exe" else ""
+          val java = s"${System.getenv("JAVA_HOME")}/bin/java${exe}"
+          val (timeType, clockType) = timeProviderType match {
+            case TimeProviderType.Static => (Some("monotonic-time"), Some("sim-clock"))
+            case TimeProviderType.WallClock => (None, None)
+          }
+          val authConfig = authSecret.fold("")(secret => s"""auth-services = [{
+                                                            |          type = unsafe-jwt-hmac-256
+                                                            |          secret = "${toJson(secret)}"
+                                                            |        }]
+                                                            |""".stripMargin)
+          val tsl = tlsConfig.fold("")(config => s"""tls {
+                 |          cert-chain-file = ${toJson(config.serverCrt)}
+                 |          private-key-file = ${toJson(config.serverPem)}
+                 |          trust-collection-file = ${toJson(config.caCrt)}
+                 |        }""".stripMargin)
+
+          def participantConfig(i: Int) = {
+            val (adminPort, ledgerApiPort) = ports(i)
+            s"""participant${i} {
+               |      admin-api.port = ${adminPort.port}
+               |      ledger-api{
+               |        port = ${ledgerApiPort.port}
+               |        ${authConfig}
+               |        ${tsl}
+               |      }
+               |      storage.type = memory
+               |      parameters = {
+               |        enable-engine-stack-traces = true
+               |        dev-version-support = ${devMode}
+               |      }
+               |      ${timeType.fold("")(x => "testing-time.type = " + x)}
+               |    }""".stripMargin
+          }
+          val participantsConfig =
+            (0 until nParticipants).map(participantConfig).mkString("\n")
+          val cantonConfig =
+            s"""canton {
+               |  parameters{
+               |    non-standard-config = yes
+               |    dev-version-support = yes
+               |    ports-file = ${toJson(portFile)}
+               |    ${clockType.fold("")(x => "clock.type = " + x)}
+               |  }
+               |
+               |  domains {
+               |    local {
+               |      storage.type = memory
+               |      public-api.port = ${domainPublicApi.port}
+               |      admin-api.port = ${domainAdminApi.port}
+               |      init.domain-parameters.protocol-version = ${if (devMode) Int.MaxValue else 4}
```

Comment:
Did you verify this works? When I used `init.domain-parameters.protocol-version` set to a number, anything that isn't 3 or 4 fails?
I had to use the word `dev`

Reply:
Did not really think about this.

I took at look a canto.
Versions are actually represented by  an Int. We have currently 4 versions
```
lazy val dev: ProtocolVersionWithStatus[Unstable] = ProtocolVersion.unstable(Int.MaxValue)
lazy val v3: ProtocolVersionWithStatus[Stable] = ProtocolVersion.stable(3)
lazy val v4: ProtocolVersionWithStatus[Stable] = ProtocolVersion.stable(4)
lazy val v5: ProtocolVersionWithStatus[Unstable] = ProtocolVersion.unstable(5)
```

However the [parser]( [canton](https://github.com/DACH-NY/canton/blob/e074aff762fb356a2fdfbeb123274776eb7693d1/community/base/src/main/scala/com/digitalasset/canton/version/ProtocolVersion.scala#L147).
) allows non Int syntax.

Reply:
I changed to "dev".